client/verta/tests/unit_tests/pipeline/test_pipeline_graph.py

```diff
@@ -0,0 +1,194 @@
+        assert graph_step.name == pipeline_step["name"]
+        assert (
+            graph_step.registered_model_version.id == pipeline_step["model_version_id"]
+        )
+        assert graph_step._registered_model.name == model_name
+        assert graph_step._registered_model.id == registered_model_id
+
+
+def test_to_graph_definition(
+    make_mock_pipeline_step, make_mock_registered_model
+) -> None:
+    """Test that a pipeline `graph` specification can be constructed from a
+    PipelineGraph object.
+    """
+    mocked_rm = make_mock_registered_model(id=123, name="test_rmv")
+    with patch.object(
+        verta.pipeline.PipelineStep, "_get_registered_model", return_value=mocked_rm
+    ):
+        step_1 = make_mock_pipeline_step("step_1")
+        step_2 = make_mock_pipeline_step("step_2")
+        step_3 = make_mock_pipeline_step("step_3")
+    step_2.set_predecessors({step_1})
+    step_3.set_predecessors({step_2})
+    graph = PipelineGraph(steps={step_1, step_2, step_3})
+    graph_spec = graph._to_graph_definition()
+    assert sorted(graph_spec, key=lambda x: x["name"]) == [
+        {
+            "name": step_1.name,
+            "predecessors": [],
+        },
+        {
+            "name": step_2.name,
+            "predecessors": [step_1.name],
+        },
+        {
+            "name": step_3.name,
+            "predecessors": [step_2.name],
+        },
+    ]
+
+
+def test_to_steps_definition(
+    make_mock_pipeline_step, make_mock_registered_model
+) -> None:
+    """Test that a pipeline `steps` specification can be constructed from a
+    PipelineGraph object.
+
+    Definitions are type list to remain json serializable.
+    """
+    mocked_rm = make_mock_registered_model(id=123, name="test_rmv")
+    with patch.object(
+        verta.pipeline.PipelineStep, "_get_registered_model", return_value=mocked_rm
+    ):
+        step_1 = make_mock_pipeline_step(name="step_1")
+        step_2 = make_mock_pipeline_step(name="step_2")
+        graph = PipelineGraph(steps={step_1, step_2})
+    step_specs = graph._to_steps_definition()
+    expected_definition = [
+        {
+            "name": step_1.name,
+            "model_version_id": step_1.registered_model_version.id,
+        },
+        {
+            "name": step_2.name,
+            "model_version_id": step_2.registered_model_version.id,
+        },
+    ]
+    assert sorted(step_specs, key=lambda x: x["name"]) == sorted(
+        expected_definition, key=lambda x: x["name"]
+    )
+
+
+def test_bad_mutation_of_step_predecessors_exception(
+    make_mock_registered_model_version,
+    make_mock_registered_model,
+    make_mock_pipeline_step,
+):
+    """Test that we throw the correct exception when a user tries to mutate
+    the predecessors of a step in an inappropriate way.
+    """
+    mocked_rmv = make_mock_registered_model_version()
+    mocked_rm = make_mock_registered_model(
+        id=mocked_rmv.registered_model_id, name="test_rmv"
```

Comment:
Maybe we should just automatically generate `name` in the fixture, as a follow-up. I shouldn't have to make this comment on every review.

```suggestion
        id=mocked_rmv.registered_model_id, name="test_rm"
```

Reply:
I think this must be CoPilot really wanting it to be "rmv" for some reason.  Sorry about that